src/components/enterprise-page/data/hooks.js

```diff
@@ -88,4 +89,47 @@ export function useEnterpriseCustomerConfig(enterpriseSlug, useCache = true) {
   }, [enterpriseSlug]);
 
   return [enterpriseConfig, fetchError];
-}
+};
+
+/**
+ * @param {string} [enterpriseId] enterprise UUID
+ * @param {object} [user] user object containing JWT roles
+ */
+export const useUpdateActiveEnterpriseForUser = ({
+  enterpriseId,
+  user,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+
+  const updateActiveEnterpriseAndRefreshJWT = useCallback(async () => {
+    setIsLoading(true);
+
+    try {
+      await updateUserActiveEnterprise(enterpriseId);
+      await loginRefresh();
+    } catch (error) {
+      logError(error);
+    } finally {
+      setIsLoading(false);
+    }
+  }, [enterpriseId]);
+
+  useEffect(() => {
+    if (!(enterpriseId && user)) {
+      return;
+    }
+
+    const { roles } = user;
+    if (roles.length > 0) {
+      // The first item in roles corresponds to the currently active enterprise for the user
+      const currentActiveEnterpriseId = roles[0].split(':')[1];
```

Comment:
What does this data object look like? An array of colon delimited strings? What other information is packed in that string?